README.md

```diff
@@ -1 +1,286 @@
+  </ul>
+</details>
+
+
+#### .get(key)
+
+Retrieve the value of the entry corresponding to `key`.
+
+#### .set(key, value)
+
+Set the value the entry corresponding to `key`.
+
+Returns itself to allow chaining.
+
+#### .tap(key, fn)
+
+Execute `fn` with the current value of the entry corresponding to `key`.   
+The return value of `fn` will replace the current value.
+
+Returns itself to allow chaining.
+
+#### .delete(key)
+
+Remove entry corresponding to `key`.
+
+Returns itself to allow chaining.
+
+#### .has(key)
+
+Check if there's an entry corresponding to `key`.
+
+Returns `true` if it exists, `false` otherwise.
+
+#### .clear()
+
+Remove all entries from the Map.
+
+Returns itself to allow chaining.
+
+#### .keys()
+
+Returns an array of the keys stored in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .values()
+
+Returns an array of the values stored in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .entries()
+
+Returns an object of all the entries stored in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .forEach(fn, [thisArg])
+
+Execute `fn` once per each key/value pair in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .merge(ob, [omit])
+
+Provide an object which maps its properties and values into the backing Map as keys and values.   
+You can also provide an array as the second argument for property names to omit from being merged.
+
+Returns itself to allow chaining.
+
+#### .extend(shorthands)
+
+Adds aliases for each `key` specified in `shorthands`.
+
+Returns itself to allow chaining.
+
+```js
+const chainedMap = new ChainedMap();
+
+chainedMap.extend(['foo', 'bar']);
+chainedMap.foo('foz');
+chainedMap.bar('baz');
+```
+
+
+### OrderableChainedMap
+
+OrderableChainedMap extends [Chainable](#chained-map) and adds a few methods to control how instances will be placed in its parent chain.
+
+<details>
+  <summary><b>Diferences from webpack-chain:</b></summary>
+  <ul>
+    <li>This class was previously named Orderable and it now extends ChainableMap</li>
+    <li>Removed weird treatment on <code>.merge(obj, [omit])</code> of <code>after</code> and <code>before</code> keys in <code>obj</code></li>
+    <li>Fixed minor bugs that caused <code>.keys()</code>, <code>.values()</code> and <code>.entries()</code> to not respect the order specified with <code>.before(key)</code> and <code>.after(key)</code>
+  </ul>
+</details>
+
```

Comment:
I think some example usages would be helpful here. It's not entirely obvious whether you can chain a `.before/.after` block and call `.end` or you need a reference to the value and just "move" it with `.before/.after`.

Reply:
Agree, I will actually add a `Examples` in each section.